javatests/google/registry/export/ExportConstantsTest.java

```diff
@@ -78,16 +91,40 @@ public void testBackupKinds_matchGoldenBackupKindsFile() throws Exception {
   }
 
   @Test
-  public void testReportingKinds_areSubsetOfBackupKinds() throws Exception {
-    assertThat(ExportConstants.getBackupKinds()).containsAllIn(ExportConstants.getReportingKinds());
+  public void testReportingKinds_matchGoldenReportingKindsFile() throws Exception {
+    URL goldenReportingKindsResource =
+        getResource(ExportConstantsTest.class, GOLDEN_REPORTING_KINDS_FILENAME);
+    final Pattern stripComments = Pattern.compile("\\s*#.*$");
```

Comment:
Rather than duplicating all the comment-stripping logic, can you just factor this out into a helper function?

```
private static ImmutableList<String> extractListFromFile(String filename) {
  String fileContents = readResourceUtf8(ExportConstantsTest.class, filename);
  final Pattern stripComments = Pattern.compile("\\s*#.*$");
  return FluentIterable.from(Splitter.on('\n').split(fileContents.trim()))
      .transform(/* comment-stripping function */)
      .toList();
}
```
Note readResourceUtf8() is a helper we've defined here:
https://github.com/google/nomulus/blob/afa4d6660176ea39efae6e9a2cc1ca915f0ba18f/java/google/registry/util/ResourceUtils.java#L35